src/api/fetch.js

```diff
@@ -0,0 +1,30 @@
+import baseURL from "./base_url";
+import paths from "./paths";
+
+export async function* indexGenerator(collection, auth, page = 1) {
+  const path = paths[collection];
+  while (path) {
```

Comment:
Wut?

Reply:
I've made it clearer that this condition should always be valid. on `.next()` the generator will pick up where it left of, so we need something to keep this looping until it reaches a `return` statement